### ISSUE TITLE
Logger outputs wrong values for different metrics

### ISSUE THREAD

**Issue by hamedhemati** [Continvvm/continuum#121]
Hi

I tried to test Continuum with a simple class incremental scenario for the Fashion MNIST dataset, but unfortunately, I could not get the correct metrics with Logger. In the code below, I am trying to test all tasks up to `task_id` with `test_task = scenario_test[:task_id]`. 

```
# Datasets
dataset_train = FashionMNIST("../../../out/data/", download=True, train=True)
dataset_test = FashionMNIST("../../../out/data/", download=True, train=False)

# Scenarios
n_way = 2
scenario_train = ClassIncremental(dataset_train, increment=n_way)
scenario_test = ClassIncremental(dataset_test, increment=n_way)

# Model
model = nn.Sequential(nn.Linear(1*28*28, 200), nn.Linear(200, 10))
optimizer = optim.SGD(model.parameters(), lr=0.001)
criterion = nn.CrossEntropyLoss()

# Training episodes
logger = Logger(list_keywords=["performance"], list_subsets=["train", "test"])
for task_id, train_task in enumerate(scenario_train):
    print("-"*20)
    print("Task: ", task_id)
    
    model.train()
    train_loader = DataLoader(train_task, batch_size=32, shuffle=True)
    # Train for one epoch
    for itr, (x,y,t) in enumerate(train_loader):
        pred = model(x.view(x.shape[0], -1))
        optimizer.zero_grad()
        loss = criterion(pred, y)
        loss.backward()
        optimizer.step()        
        pred_ = torch.argmax(pred, dim=1).detach()
        logger.add([pred_, y, t], subset="train", keyword="performance")
   
    # Test all tasks up to task task_id
    test_task = scenario_test[:task_id]
    test_loader = DataLoader(test_task, batch_size=32, shuffle=True)
    model.eval()
    for itr, (x, y, t) in enumerate(test_loader):
        pred = model(x.view(x.shape[0], -1))
        pred_ = torch.argmax(pred, dim=1).detach()
        logger.add([pred_, y, t], subset="test", keyword="performance")

    print(f"acc: {logger.accuracy}, online acc: {logger.average_incremental_accuracy} ")
    print(f"BWT: {logger.backward_transfer}, FWT: {logger.forward_transfer}")
    logger.end_task()

```

And this is how the output looks like:

```
--------------------
Task:  0
acc: 0.9385, online acc: 0.9385 
BWT: 0.9385, FWT: 0.9385
--------------------
Task:  1
acc: 0.9385, online acc: 0.9385 
BWT: 0.9385, FWT: 0.9385
--------------------
Task:  2
acc: 0.9385, online acc: 0.9385 
BWT: 0.9385, FWT: 0.9385
--------------------
Task:  3
acc: 0.9385, online acc: 0.9385 
BWT: 0.9385, FWT: 0.9385
--------------------
Task:  4
acc: 0.9385, online acc: 0.9385 
BWT: 0.9385, FWT: 0.9385
```
I looked into the implementation of the classes `Logger` and `_BaseLogger` and it seems like directly using the `accuracy` method from `continuum.metrics.metrics` works properly when used in a similar way as in the `accuracy` method of the class `Logger`. I think it might have something to do with the `@cache` decorators used in the class Logger as it does not retrieve the current predictions and targets. 
For the backward and forward transfer, I also tried to use the ` backward_transfer` from `continuum.metrics.metrics` directly, but I would get nan after 3 episodes. It seems like `logger.add([pred, y, t], keyword="performance")` does not add the task ids `t` because in its implementation in the `_BaseLogger` class it check `if (task_ids is not None) and self._get_current_task_ids(subset).size > 0:`  and since task_ids is empty at the beginning of training it remains empty for the rest of the episodes.  

Am I doing something wrong with regard to the usage of  Logger?

Thanks :)

**Comment by arthurdouillard** [comment]
Hello! Sorry for the inconvenience and thank you for the bug report.

There was indeed a problem with the cache which appeared during a recent refactor. I have disabled the cache for now.
Furthermore, you made a tiny mistake in your code, by doing  `scenario_test[:task_id]`, you should do `scenario_test[:task_id + 1]`. I have add an error so that it can be more explicit.

Everything has been fixed in #122, and you can update your continuum to 1.0.13.

Now I get those metrics results:
```
--------------------
Task:  0
acc: 0.947, online acc: 0.947 
BWT: 0.0, FWT: 0.0
--------------------
Task:  1
acc: 0.4795, online acc: 0.7132499999999999 
BWT: 0.0, FWT: 0.0
--------------------
Task:  2
acc: 0.3303333333333333, online acc: 0.5856111111111111 
BWT: -0.31566666666666665, FWT: 0.0
--------------------
Task:  3
acc: 0.249375, online acc: 0.5015520833333333 
BWT: -0.4755, FWT: 0.0
--------------------
Task:  4
acc: 0.1988, online acc: 0.4410016666666666 
BWT: -0.575, FWT: 5e-05
```

(bwt and fwt only starts at the third task, it's their normal behavior)
